### ISSUE TITLE
Add mirroring of slabs to CoherentInterfaceBuilder

### ISSUE THREAD

**Issue by genijalac11** [materialsproject/pymatgen#2406]
The _CoherentInterfaceBuilder_ works rather nicely when creating interfaces between various lattices. However, it seems like it currently doesn't consider mirroring non-symmetric slabs, which then leaves quite a few possible structures unexplored. 
Not sure whether this would be easier to implement at this step or in the _Surface_ module? 
(If that feature is already present and I missed it, let me know!)

**Comment by arosen93** [comment]
For what it's worth, I think this kind of feature would be desirable (and is not currently present) for virtually all the slab functions. Perhaps a better solution might be to have a `slab_flip()` function in pymatgen that will: 1) detect if a slab is asymmetric; 2) provide the flipped version (with the appropriate constraints still present). This could then be called after any kind of slab generation scheme to augment the returned set of objects.

**Comment by shyuep** [comment]
Yes this would be a helpful feature! But if you mirror, wouldn't that create a twin boundary?

**Comment by genijalac11** [comment]
Hmm I'm not sure that would happen. What the _CoherentInterfaceBuilder_ module does now is to take two specified slabs and make an interface out of them. If both slabs are symmetric, there is no issue with the orientation of each. However, if the surfaces are non-symmetric there is a number of combinations between the distinct surface terminations. So far the module does not take that into account and simply aligns whatever is the top of the substrate with the bottom of the film (oriented along the non periodic direction).
